device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1296,21 +1304,24 @@ else if (transportException != null && !transportException.isRetryable())
     // Still need to check the device connection status before you can report the device to be re-connected.
     private void singleDeviceReconnectAttemptAsync(String deviceId)
     {
-        DeviceClientConfig config = getConfig(deviceId);
+        DeviceClientConfig config = this.getConfig(deviceId);
+
+        if (config == null)
+        {
+            log.debug("Reconnection for device {} was abandoned because it was unregistered while reconnecting", deviceId);
+            return;
+        }
+
         ((AmqpsIotHubConnection) this.iotHubTransportConnection).unregisterMultiplexedDevice(config);
         ((AmqpsIotHubConnection) this.iotHubTransportConnection).registerMultiplexedDevice(config);
     }
 
     private DeviceClientConfig getConfig(String deviceId)
     {
-        DeviceClientConfig config = this.deviceClientConfigs.get(deviceId);
-
-        if (config != null)
-        {
-            return config;
-        }
-
-        throw new IllegalStateException(String.format("Device client config does not exist for device %s", deviceId));
```

Comment:
This method just returns a null value rather than throw an exception now. Each method that calls this method will handle this case a little differently (abandoning reconnection, for instance)